x-pack/plugins/session_view/public/components/session_leader_table/index.tsx

```diff
@@ -0,0 +1,324 @@
+            defaultMessage: 'Expand',
+          })}
+          color="primary"
+          iconType="expand"
+          onClick={() => onExpand(actionProps)}
+        />
+      </EuiToolTip>
+    );
+  };
+
+  const renderInspectButton = (actionProps: ActionProps) => {
+    return (
+      <EuiToolTip position="top" content="Inspect">
+        <EuiButtonIcon
+          aria-label={i18n.translate('xpack.sessionView.sessionLeaderTable.inspect', {
+            defaultMessage: 'Inspect',
+          })}
+          color="primary"
+          iconType="inspect"
+          onClick={() => onInspect(actionProps)}
+        />
+      </EuiToolTip>
+    );
+  };
+
+  const renderOpenMoreActionsButton = (actionProps: ActionProps) => {
+    const { eventId } = actionProps;
+    return (
+      <EuiPopover
+        anchorPosition="upCenter"
+        isOpen={openPopoverId === eventId}
+        closePopover={handleClosePopover}
+        button={
+          <EuiButtonIcon
+            aria-label={i18n.translate('xpack.sessionView.sessionLeaderTable.openInSessionView', {
+              defaultMessage: 'Open in Session View',
+            })}
+            color="primary"
+            iconType="boxesHorizontal"
+            onClick={handleMoreActionsClick(eventId)}
+            data-test-subj={`session-leader-table-more-actions-${eventId}`}
+          />
+        }
+      >
+        <EuiContextMenuPanel
+          size="s"
+          items={[
+            <EuiContextMenuItem
+              key="analyzeSession"
+              onClick={() => {
+                onAnalyzeSession(actionProps);
+                handleClosePopover();
+              }}
+            >
+              {i18n.translate('xpack.sessionView.sessionLeaderTable.analyzeSession', {
+                defaultMessage: 'Analyze Session',
+              })}
+            </EuiContextMenuItem>,
+            <EuiContextMenuItem
+              key="openSessionViewer"
+              onClick={() => {
+                onOpenSessionViewer(actionProps);
+                handleClosePopover();
+              }}
+            >
+              {i18n.translate('xpack.sessionView.sessionLeaderTable.openInSessionView', {
+                defaultMessage: 'Open in Session View',
+              })}
+            </EuiContextMenuItem>,
+          ]}
+        />
+      </EuiPopover>
+    );
+  };
+
+  const renderLeadingControlCell = (actionProps: ActionProps) => {
+    return (
+      <div css={rowButtonContainer}>
+        {renderRowCheckbox(actionProps)}
+        {renderExpandButton(actionProps)}
+        {renderInspectButton(actionProps)}
+        {renderOpenMoreActionsButton(actionProps)}
+      </div>
+    );
+  };
+
+  const renderCellValue = ({ columnId, data }: CellValueElementProps) => {
+    const value = data.find((o) => o.field === columnId)?.value?.[0];
+    return value || <>&#8212;</>;
+  };
+
+  const renderUnit = () => {
+    return 'events';
+  };
+
+  const renderTimelinesTable = () => {
```

Comment:
will also have the added benefit of being able to easily use filters/query bar/interact with events in the other views which would be harder with how it works now

Reply:
Since we will have to add more changes to the Session Leader Table, we will remove it from the first PR, and introduce Session Leader Table in separate PRs alongside these suggestions for better integration support!